Sources/Classes/RSEventRepository.m

```diff
@@ -560,5 +571,24 @@ - (void) __prepareScreenRecorder {
     [UIViewController rudder_swizzleView];
 }
 
+- (void) registerBackGroundTask {
+    [RSLogger logDebug:@"EventRepository: registerBackGroundTask: Registering for Background run time"];
+    backgroundTask = [[UIApplication sharedApplication] beginBackgroundTaskWithExpirationHandler:^{
+        [self endBackGroundTask];
+    }];
+}
+
+- (void) endBackGroundTask {
+    [[UIApplication sharedApplication] endBackgroundTask:backgroundTask];
+    backgroundTask = UIBackgroundTaskInvalid;
+}
+
+- (void) reInstateBackGroundTask {
```

Comment:
Can we merge this function with `registerBackGroundTask`? Make 1 function to `registerBackGroundTask`.
```
if(backgroundTask != UIBackgroundTaskInvalid) {
        [self endBackGroundTask];
    } else {
backgroundTask = [[UIApplication sharedApplication] beginBackgroundTaskWithExpirationHandler:^{
        [self endBackGroundTask];
    }];
}
```

Reply:
Yeah, we can do that